src/controller.cpp

```diff
@@ -617,8 +599,11 @@ void Controller::activateNext() {
   if (m_initiator == ExtensionUser) {
     return;
   }
-  // Move to the confirming state if we are awaiting any connection handshakes.
-  setState(StateConfirming);
+
+  if (m_state != StateSilentSwitching) {
```

Comment:
This makes sense generally, but can you help me understand what would have happened if we didn't wrap this in the if statement?

Reply:
Sure, if we change the state here during a silent server switch we will see the 'switching' UI on the front-end and potentially cause confusion for the subscriber.